module/mempool/herocache/backdata/heropool/pool_test.go

```diff
@@ -456,7 +456,7 @@ func testAddingEntities(t *testing.T, pool *Pool, entitiesToBeAdded []*unittest.
 
 			if i >= len(pool.poolEntities) {
 				require.True(t, slotAvailable)
-				require.True(t, ejectionHappened)
+				require.NotNil(t, ejectedEntity)
```

Comment:
To ensure the reliability of the process, in `LRUEjection` mode, we should implement a more rigorous check to validate both the correctness and uniqueness of the `ejectedEntity`. Specifically, we need to confirm that (1) the `ejectedEntity` indeed belongs to the `entitiesToBeAdded`, and (2) it is the oldest entity among the ones in the pool. To achieve this, we can establish a condition that guarantees the uniqueness of all `entitiesToBeAdded` so that the `ejectedEntity` will also be unique. Simply relying on the `require.NotNil(t, ejectedEntity)` statement is not sufficient, as this method fails to detect potential issues for example if we happen to return a random mock entity every time we eject.

Reply:
totally makes sense, updated: https://github.com/onflow/flow-go/pull/3961/commits/9f4d76d8ab4fb02b54a5ecccc58c303944ffb50f